Content.Client/Items/Managers/ItemSlotManager.cs

```diff
@@ -84,21 +84,15 @@ public bool OnButtonPressed(GUIBoundKeyEventArgs args, IEntity? item)
                 _entitySystemManager.GetEntitySystem<VerbSystem>()
                                     .OpenContextMenu(item, _uiMgr.ScreenToUIPosition(args.PointerLocation));
             }
-            else if (args.Function == ContentKeyFunctions.ActivateItemInWorld)
+            else if (args.Function == ContentKeyFunctions.ActivateItemInWorld || args.Function == ContentKeyFunctions.AltUse)
             {
                 var inputSys = _entitySystemManager.GetEntitySystem<InputSystem>();
 
                 var func = args.Function;
                 var funcId = _inputManager.NetworkBindMap.KeyFunctionID(args.Function);
 
-
-                var mousePosWorld = _eyeManager.ScreenToMap(args.PointerLocation);
-
-                var coordinates = _mapManager.TryFindGridAt(mousePosWorld, out var grid) ? grid.MapToGrid(mousePosWorld) :
-                    EntityCoordinates.FromMap(_mapManager, mousePosWorld);
-
                 var message = new FullInputCmdMessage(_gameTiming.CurTick, _gameTiming.TickFraction, funcId, BoundKeyState.Down,
-                    coordinates, args.PointerLocation, item.Uid);
+                    item.Transform.Coordinates, args.PointerLocation, item.Uid);
```

Comment:
I dont rightly know what I did here... but it works? Someone who knows more about coordinates and maps should check this. I just stole ther working code from `ContextMenuPresenter.OnKeyBindDownStack()` which perfomed a similar operation

The old code fails the user input validation check in InteractionSystem with the console showing:
```
[INFO] system.interaction: Invalid Coordinates: client=localhost@ElectroSR, coords=EntId=c1, X=0.00, Y=0.00
[INFO] system.interaction: ActivateItemInWorld input validation failed
```

Reply:
the previous code is trying to get the literal spot that the player clicked by going screencoords -> mapcoords -> entitycoords, this code just gets the position of the entity they clicked on

this might mean that after sending the message the server might be incorrectly validating the click location? idk exactly what this code handles

Reply:
Yeah this doesn't seem like the solution. Even if it were sending the coordinates in addition to the item is silly because now you have ping being a problem.

Reply:
I'm really not sure WHY this input is generating a new input command for inputSys.HandleInputCommand() in the first place. Shouldn;t it instead just be doing the interaction (which I think in this case is raising an interaction event for the server to handle)?

Reply:
A bit more on what was going wrong: `_eyeManager.ScreenToMap(args.PointerLocation)`; returns default/null coordinates when clicking on UI elements because its not clicking on a view-port. So that is part of why it wasn't working.

Generating a new mouse click for the server to handle shouldn't really cause more latency, because the initial click is all handled client-side and it needs to be sent to the server anyway. So currently just directly using the coordinates of the entity that was clicked on to generate a new click for the server to handle works, but is a really weird way of doing it. It should just be sending networked events to the interaction system, and not using fake inputs to communicate.

Reply:
I've replaced this "fake input" with a network event that InteractionSystem subscribes to.